packages/flutter_tools/lib/src/commands/symbolize.dart

```diff
@@ -108,47 +151,71 @@ class SymbolizeCommand extends FlutterCommand {
       input = _stdio.stdin;
     }
 
-    String debugInfoPath = stringArg('debug-info')!;
-
-    // If it's a dSYM container, expand the path to the actual DWARF.
-    if (debugInfoPath.endsWith('.dSYM')) {
-      final Directory debugInfoDir = _fileSystem
-        .directory(debugInfoPath)
-        .childDirectory('Contents')
-        .childDirectory('Resources')
-        .childDirectory('DWARF');
-
-      final List<FileSystemEntity> dwarfFiles = debugInfoDir.listSync().whereType<File>().toList();
-      if (dwarfFiles.length == 1) {
-        debugInfoPath = dwarfFiles.first.path;
-      } else {
-        throwToolExit('Expected a single DWARF file in a dSYM container.');
+    final Map<int, String> debugInfoPathMap = _unitDebugInfoPathMap();
+    final Map<int, Uint8List> unitSymbols = <int, Uint8List>{};
+
+    for (final int unitId in debugInfoPathMap.keys) {
+      String debugInfoPath = debugInfoPathMap[unitId]!;
+
+      // If it's a dSYM container, expand the path to the actual DWARF.
+      if (debugInfoPath.endsWith('.dSYM')) {
+        final Directory debugInfoDir = _fileSystem
+          .directory(debugInfoPath)
+          .childDirectory('Contents')
+          .childDirectory('Resources')
+          .childDirectory('DWARF');
+
+        final List<FileSystemEntity> dwarfFiles = debugInfoDir.listSync().whereType<File>().toList();
+        if (dwarfFiles.length == 1) {
+          debugInfoPath = dwarfFiles.first.path;
+        } else {
+          throwToolExit('Expected a single DWARF file in a dSYM container.');
```

Comment:
This is maybe a philosophical question, but I'd prefer this not be a ToolExit. `ToolExit` means user error that only they can fix. It's not clear to me if they hit this point, that it is necessarily their fault, nor that they would know what to do (if I hit this, I'd be like, "what does that mean?").

Also ToolExits don't show up in our crash logging, whereas if lots of users hit this we would probably want to know about this. What do you think?

Reply:
Yeah, you're right, that wouldn't be ideal in either situation then. What would be the best approach for that case? Is there something better for revealing internal errors (e.g., something that's our fault) or should it just `throw`?

Reply:
Have just used `.single` so that it causes an appropriate `StateError` if it turns out the directory isn't populated by a single file (either empty or more than one).